### ISSUE TITLE
Skinned mesh / actor triggers an invalid memory access

### ISSUE THREAD

**Issue by realJavabot** [o3de/o3de#18027]
**Describe the bug**
When working on a level, if I have a Actor it can crash the editor when I press play or if I add a lua script to the actor. This does not always happen, but it seems to occur several minutes after I create the actor or load a level with the actor. There hasn't been any changes to this file in like a year, so this may be something to do with the FBX itself.

There is a static cast that happens implicitly from `GetBufferAsset()->`, so this is probably the part of this line triggering the error.

**Assets required**
[low_poly_human_fbx.txt](https://github.com/user-attachments/files/15795365/low_poly_human_fbx.txt)
[low_poly_human.fbx_assetinfo.txt](https://github.com/user-attachments/files/15795366/low_poly_human.fbx_assetinfo.txt)

**Steps to reproduce**
I'm not sure exactly what causes this, but having the actor in the scene and working on it in the editor eventually leads to this exception showing up when attempting game mode.

**Expected behavior**
no exception

**Actual behavior**
exception

**Screenshots/Video**
![exception](https://github.com/o3de/o3de/assets/21233659/93c0dcbd-b383-4c40-846e-b41eeedb107a)
![callstack](https://github.com/o3de/o3de/assets/21233659/a70cbdaf-4d95-4f3c-9b3a-efcf0038ca43)
![locals](https://github.com/o3de/o3de/assets/21233659/99c74728-f279-4afa-a108-0477978e2b8d)

**Found in Branch**
Development branch

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
P8e66aced697f18e82afeb2cadb7974537202f73e

**Desktop/Device (please complete the following information):**
    Device: PC
    OS: Windows
    Version 10
    CPU Ryzen 3600
    GPU GTX 1070
    Memory 32GB

**Additional context**

**Comment by moudgils** [comment]
Looks like the code needs to handle incorrect or incomplete assets gracefully. Are you able to add a null check, possibly add an assert and exit out of the method to see if that fixes the crash itself. It is possible that the asset may not render correctly. The hope here is that the developer will see that the asset is missing information and address that as needed.

**Comment by moudgils** [comment]
Tagging @nick-l-o3de as you were also looking at similar issue whereby creating skinned mesh code doesnt handle bad/incomplete data properly.

**Comment by realJavabot** [comment]
Wanted to add that this can also happen if your mesh has an object not part of the skin.

**Comment by realJavabot** [comment]
The initial load appears to handle the unloaded asset correctly, but it causes some issues immediately after:
```
<21:49:26> Trace::Assert
 D:\o3de\Code\Framework\AzCore\AzCore/Asset/AssetCommon.h(339): (34440) 'class AZ::RPI::BufferAsset *__cdecl AZ::Data::Asset<class AZ::RPI::BufferAsset>::operator ->(void) const'

<21:49:26> Asset {4CE2FA19-4EF1-5C82-89CF-0EC68ABC0E44}:10ef8d28 (assets/low_poly_human_lod0_skin_jointindices0.fbx.azbuffer) is not loaded

<21:49:26> ------------------------------------------------

<21:49:28> D:\o3de\Code\Framework\AzCore\AzCore\Asset\AssetCommon.h (339) : AZ::Data::Asset<AZ::RPI::BufferAsset>::operator->

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\ActorAsset.cpp (215) : AZ::Render::CreateSkinnedMeshInputFromActor

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActor.cpp (50) : AZ::Render::AtomActor::FindOrCreateSkinnedMeshInputBuffers

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (456) : AZ::Render::AtomActorInstance::Create

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (193) : AZ::Render::AtomActorInstance::Activate

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (62) : AZ::Render::AtomActorInstance::AtomActorInstance

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomBackend.cpp (35) : AZ::Render::AtomBackend::CreateActorInstance

<21:49:28> D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (1018) : EMotionFX::Integration::EditorActorComponent::CheckActorCreation

<21:49:28> D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (700) : EMotionFX::Integration::EditorActorComponent::OnTick

<21:49:28> D:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::ScriptTi

<21:49:28> D:\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1645) : AZ::ComponentApplication::Tick

<21:49:28> D:\o3de\Code\Editor\GameEngine.cpp (765) : CGameEngine::Update

<21:49:28> D:\o3de\Code\Editor\CryEdit.cpp (2256) : CCryEditApp::IdleProcessing

<21:49:28> D:\o3de\Code\Editor\CryEdit.cpp (2153) : CCryEditApp::OnIdle

<21:49:28> D:\o3de\Code\Editor\Core\QtEditorApplication.cpp (319) : Editor::EditorQtApplication::maybeProcessIdle

<21:49:28> 00007FF919FCF72A (Qt5Core) : QTimer::timeout

<21:49:28> 00007FF919FC82DD (Qt5Core) : QObject::event

<21:49:28> 00007FF9355449A1 (Qt5Widgets) : QApplicationPrivate::notify_helper

<21:49:28> 00007FF935543A23 (Qt5Widgets) : QApplication::notify

<21:49:28> 00007FF919FA17CB (Qt5Core) : QCoreApplication::notifyInternal2

<21:49:28> 00007FF919FEC1B9 (Qt5Core) : QEventDispatcherWin32Private::sendTimerEvent

<21:49:28> 00007FF919FEA5E9 (Qt5Core) : QEventDispatcherWin32::event

<21:49:28> 00007FF9355449A1 (Qt5Widgets) : QApplicationPrivate::notify_helper

<21:49:28> 00007FF935543A23 (Qt5Widgets) : QApplication::notify

<21:49:28> 00007FF919FA17CB (Qt5Core) : QCoreApplication::notifyInternal2

<21:49:28> ==================================================================

<21:49:28> ====Assert added to ignore list by spec and verbosity setting.====

<21:49:28> (Exit) - Exception with exit code: 0xc0000005
<21:49:28> (Exit) - D:\o3de\Code\Legacy\CrySystem\DebugCallStack.cpp (229) : DebugCallStack::handleException

<21:49:28> (Exit) - 00007FF9EF2FD827 (KERNELBASE) : UnhandledExceptionFilter

<21:49:28> (Exit) - 00007FF9F1535898 (ntdll) : memset

<21:49:28> (Exit) - 00007FF9F151CE56 (ntdll) : _C_specific_handler

<21:49:28> (Exit) - 00007FF9F153292F (ntdll) : _chkstk

<21:49:28> (Exit) - 00007FF9F14E2554 (ntdll) : RtlRaiseException

<21:49:28> (Exit) - 00007FF9F153143E (ntdll) : KiUserExceptionDispatcher

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\ActorAsset.cpp (215) : AZ::Render::CreateSkinnedMeshInputFromActor

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActor.cpp (50) : AZ::Render::AtomActor::FindOrCreateSkinnedMeshInputBuffers

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (456) : AZ::Render::AtomActorInstance::Create

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (193) : AZ::Render::AtomActorInstance::Activate

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (62) : AZ::Render::AtomActorInstance::AtomActorInstance

<21:49:28> (Exit) - D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomBackend.cpp (35) : AZ::Render::AtomBackend::CreateActorInstance

<21:49:28> (Exit) - D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (1018) : EMotionFX::Integration::EditorActorComponent::CheckActorCreation

<21:49:28> (Exit) - D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (700) : EMotionFX::Integration::EditorActorComponent::OnTick

<21:49:28> (Exit) - D:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::ScriptTi

<21:49:28> (Exit) - D:\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1645) : AZ::ComponentApplication::Tick

<21:49:28> (Exit) - D:\o3de\Code\Editor\GameEngine.cpp (765) : CGameEngine::Update

<21:49:28> (Exit) - D:\o3de\Code\Editor\CryEdit.cpp (2256) : CCryEditApp::IdleProcessing

<21:49:28> (Exit) - D:\o3de\Code\Editor\CryEdit.cpp (2153) : CCryEditApp::OnIdle

<21:49:28> (Exit) - D:\o3de\Code\Editor\Core\QtEditorApplication.cpp (319) : Editor::EditorQtApplication::maybeProcessIdle

<21:49:28> (Exit) - 00007FF919FCF72A (Qt5Core) : QTimer::timeout

<21:49:28> (Exit) - 00007FF919FC82DD (Qt5Core) : QObject::event

<21:49:28> (Exit) - 00007FF9355449A1 (Qt5Widgets) : QApplicationPrivate::notify_helper

<21:49:28> (Exit) - 00007FF935543A23 (Qt5Widgets) : QApplication::notify

<21:49:28> 00007FF9EF2FD827 (KERNELBASE) : UnhandledExceptionFilter

<21:49:28> 00007FF9F1535898 (ntdll) : memset

<21:49:28> 00007FF9F151CE56 (ntdll) : _C_specific_handler

<21:49:28> 00007FF9F153292F (ntdll) : _chkstk

<21:49:28> 00007FF9F14E2554 (ntdll) : RtlRaiseException

<21:49:28> 00007FF9F153143E (ntdll) : KiUserExceptionDispatcher

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\ActorAsset.cpp (215) : AZ::Render::CreateSkinnedMeshInputFromActor

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActor.cpp (50) : AZ::Render::AtomActor::FindOrCreateSkinnedMeshInputBuffers

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (456) : AZ::Render::AtomActorInstance::Create

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (193) : AZ::Render::AtomActorInstance::Activate

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomActorInstance.cpp (62) : AZ::Render::AtomActorInstance::AtomActorInstance

<21:49:28> D:\o3de\Gems\AtomLyIntegration\EMotionFXAtom\Code\Source\AtomBackend.cpp (35) : AZ::Render::AtomBackend::CreateActorInstance

<21:49:28> D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (1018) : EMotionFX::Integration::EditorActorComponent::CheckActorCreation

<21:49:28> D:\o3de\Gems\EMotionFX\Code\Source\Integration\Editor\Components\EditorActorComponent.cpp (700) : EMotionFX::Integration::EditorActorComponent::OnTick

<21:49:28> D:\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::ScriptTi

<21:49:28> D:\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1645) : AZ::ComponentApplication::Tick

<21:49:28> D:\o3de\Code\Editor\GameEngine.cpp (765) : CGameEngine::Update

<21:49:28> D:\o3de\Code\Editor\CryEdit.cpp (2256) : CCryEditApp::IdleProcessing

<21:49:28> D:\o3de\Code\Editor\CryEdit.cpp (2153) : CCryEditApp::OnIdle

<21:49:28> D:\o3de\Code\Editor\Core\QtEditorApplication.cpp (319) : Editor::EditorQtApplication::maybeProcessIdle

<21:49:28> 00007FF919FCF72A (Qt5Core) : QTimer::timeout

<21:49:28> 00007FF919FC82DD (Qt5Core) : QObject::event

<21:49:28> 00007FF9355449A1 (Qt5Widgets) : QApplicationPrivate::notify_helper

<21:49:28> 00007FF935543A23 (Qt5Widgets) : QApplication::notify

<21:49:28> 00007FF919FA17CB (Qt5Core) : QCoreApplication::notifyInternal2```

**Comment by nick-l-o3de** [comment]
This is possibly fixed by https://github.com/o3de/o3de/pull/18056 - I will retest once that ends up in development.

**Comment by nick-l-o3de** [comment]
This is not fixed by 18056, there are other related issues.

**Comment by nick-l-o3de** [comment]
I reproed it by duplicating the entity, still investigating.

**Comment by nick-l-o3de** [comment]
So this happens only when an asset is reloaded.  The Actor asset has a weird way of reloading, in that it ignores the new actor asset handed to it in the OnAssetReloaded() call, and instead, calls DestroyActorInstance and m_actorAsset.Release() and then waits for the Asset Unloaded function to be called.  DestroyActorInstance also deletes the render actor instance, in which all the 

This is flawed in a number of ways, not least of which, Other things might have refs to the old data (who knows what - previewers, emotionfx itself, things that are not also EditorActorComponents) and will not release their refs, which will cause the data to not be Unloaded, ever.  

But assuming that doesn't happen, and everything in the entire system drops its refcounted refs on the AZ::Asset<EmotionFXActor>,  all existing `EditorActorComponent` in the scene will get an `OnAssetUnloaded` on the same frame.

They will get all the direct dependencies of the actor asset, and find the ModelAsset (as in, atom model).  They will then call ReloadModel (which thankfully internally de-duplicates the reload calls), and starts the reload, which seems to be quite a complex system (see `ModelReloader.cpp`) doing things in layers.  Presumably the job here is to finish loading everything, before the callback to m_actorAsset.QueueLoad() is invoked.  

But somehow, when we get here, instead, _most_ of the model data is loaded, but the stream buffers for the lods are not.  I'm not sure why this only happens on reload, but theres a few things that could be done here.

One would be to make it so that the actor forbids auto-reload of its actor asset time by updating the actor asset handler to return false for the "can I automatically reload" bool override.  Since it clearly wants to handle itself.  Then the actor components can listen to the "asset changed on disk" bus instead, and do the same thing (that is, drop their ref and load a new one) so that we aren't wasting loads. Unless we do this, then whats going to happen is that the new actor will actually automatically reload and only after its fully finished loading, does it offer it to you in OnAssetReloaded() which is a waste, since it dose not pin the newly loaded one.

Secondly, we need to figure out why the model lods streambuffer assets are not loaded on reload.  They are not invalid - they have a valid asset name, assetid, state (its "not loaded"), its just... well, not loaded at all at reload time.  Perhaps a bug with the AssetReloader?

**Comment by nick-l-o3de** [comment]
Further information here, its nothing to do with reload.
easy repro 100% is 
1. load a scene up (it can becompletely empty)
2. drop any actor in
3. Open the FBX settings panel for the actor's asset
5. Hit save
6. Duplicate the actor

It appears to be becuase the *PREVIEW WIDGET* registers the mesh with the mesh feature processor, and does so in a way that doesn't ask the feature processor to preserve the buffer assets (ie, the lod's tangents, etc).  The mesh feature processor drops the ref of those assets to 0 and they unload.

The next time you duplicate that actor, it is missing the data.  Note that its still a mystery why - the actor calls `AcquireMesh` on the feature processor with a flag that makes it keep the buffer, so its refcount should increase each actor in the scene, but somehow, when the previewer gets it, its refcount goes from 1 to 0, rather than 2 to 1.
